contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -275,27 +283,42 @@ func taskRunning(t *Task) taskStateFn {
 	}()
 
 	// wait for the process to complete, or else for a "quit" signal on the task (at which point we'll attempt to kill manually)
-	select {
-	case <-t.shouldQuit:
-		// check for tie
+	gracePeriod := defaultKillGracePeriod
+	forceKill := false
+	wr := &Completion{err: fmt.Errorf("failed to kill process: %q", t.bin)}
+	defer func() { trySend(wr) }() // don't capture wr early
+waitLoop:
+	for i := 0; i < 2; i++ {
```

Comment:
pretty deep nesting with this for-loop. At least add a comment why we do this twice.